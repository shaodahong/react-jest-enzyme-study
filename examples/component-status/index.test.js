import React from 'react';
import ComponentState from './index';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('ComponentStatus test', () => {
  it('ComponentStatus snap', () => {
    const wrapper = shallow(<ComponentState />);
    expect(wrapper).toMatchSnapshot();
  });
  
  it('ComponentStatus props', () => {
    const wrapper = shallow(<ComponentState ext="shanghai" />);
  
    expect(wrapper.text()).toBe('hello shanghai');
  });
});
