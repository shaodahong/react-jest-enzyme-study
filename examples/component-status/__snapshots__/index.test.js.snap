// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentStatus test ComponentStatus snap 1`] = `
<div>
  hello
</div>
`;
